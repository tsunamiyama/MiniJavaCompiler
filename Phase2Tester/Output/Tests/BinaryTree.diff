--- SelfTestCases/BinaryTree.out	2011-04-28 14:55:20.000000000 -0700
+++ ./Output/Tests/BinaryTree.out	2021-02-13 18:30:49.235414500 -0800
@@ -15,12 +15,12 @@
 1
 0
 1
-4
 8
+12
 14
 16
 20
 24
 28
-0
+1
 0
