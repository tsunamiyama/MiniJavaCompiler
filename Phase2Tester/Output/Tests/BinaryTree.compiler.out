const vmt_BT
	:BT.Start

const vmt_Tree
	:Tree.Init
	:Tree.SetRight
	:Tree.SetLeft
	:Tree.GetRight
	:Tree.GetLeft
	:Tree.GetKey
	:Tree.SetKey
	:Tree.GetHas_Right
	:Tree.GetHas_Left
	:Tree.SetHas_Left
	:Tree.SetHas_Right
	:Tree.Compare
	:Tree.Insert
	:Tree.Delete
	:Tree.Remove
	:Tree.RemoveRight
	:Tree.RemoveLeft
	:Tree.Search
	:Tree.Print
	:Tree.RecPrint

func Main()
	t.0 = HeapAllocZ(4)
	[t.0] = :vmt_BT
	t.1 = t.0
	if t.1 goto :null1
		Error("null pointer")
	null1:
	t.2 = [t.1]
	t.2 = [t.2+0]
	t.3 = call t.2(t.1)
	t.4 = t.3
	PrintIntS(t.4)
	ret

func BT.Start(this)
	t.0 = HeapAllocZ(28)
	[t.0] = :vmt_Tree
	t.1 = t.0
	t.2 = t.1
	root = t.2
	t.3 = root
	t.4 = 16
	t.5 = t.4
	if t.3 goto :null2
		Error("null pointer")
	null2:
	t.6 = [t.3]
	t.6 = [t.6+0]
	t.7 = call t.6(t.3 t.5)
	t.8 = t.7
	ntb = t.8
	t.9 = root
	if t.9 goto :null3
		Error("null pointer")
	null3:
	t.10 = [t.9]
	t.10 = [t.10+72]
	t.11 = call t.10(t.9)
	t.12 = t.11
	ntb = t.12
	t.13 = 100000000
	t.14 = t.13
	PrintIntS(t.14)
	t.15 = root
	t.16 = 8
	t.17 = t.16
	if t.15 goto :null4
		Error("null pointer")
	null4:
	t.18 = [t.15]
	t.18 = [t.18+48]
	t.19 = call t.18(t.15 t.17)
	t.20 = t.19
	ntb = t.20
	t.21 = root
	if t.21 goto :null5
		Error("null pointer")
	null5:
	t.22 = [t.21]
	t.22 = [t.22+72]
	t.23 = call t.22(t.21)
	t.24 = t.23
	ntb = t.24
	t.25 = root
	t.26 = 24
	t.27 = t.26
	if t.25 goto :null6
		Error("null pointer")
	null6:
	t.28 = [t.25]
	t.28 = [t.28+48]
	t.29 = call t.28(t.25 t.27)
	t.30 = t.29
	ntb = t.30
	t.31 = root
	t.32 = 4
	t.33 = t.32
	if t.31 goto :null7
		Error("null pointer")
	null7:
	t.34 = [t.31]
	t.34 = [t.34+48]
	t.35 = call t.34(t.31 t.33)
	t.36 = t.35
	ntb = t.36
	t.37 = root
	t.38 = 12
	t.39 = t.38
	if t.37 goto :null8
		Error("null pointer")
	null8:
	t.40 = [t.37]
	t.40 = [t.40+48]
	t.41 = call t.40(t.37 t.39)
	t.42 = t.41
	ntb = t.42
	t.43 = root
	t.44 = 20
	t.45 = t.44
	if t.43 goto :null9
		Error("null pointer")
	null9:
	t.46 = [t.43]
	t.46 = [t.46+48]
	t.47 = call t.46(t.43 t.45)
	t.48 = t.47
	ntb = t.48
	t.49 = root
	t.50 = 28
	t.51 = t.50
	if t.49 goto :null10
		Error("null pointer")
	null10:
	t.52 = [t.49]
	t.52 = [t.52+48]
	t.53 = call t.52(t.49 t.51)
	t.54 = t.53
	ntb = t.54
	t.55 = root
	t.56 = 14
	t.57 = t.56
	if t.55 goto :null11
		Error("null pointer")
	null11:
	t.58 = [t.55]
	t.58 = [t.58+48]
	t.59 = call t.58(t.55 t.57)
	t.60 = t.59
	ntb = t.60
	t.61 = root
	if t.61 goto :null12
		Error("null pointer")
	null12:
	t.62 = [t.61]
	t.62 = [t.62+72]
	t.63 = call t.62(t.61)
	t.64 = t.63
	ntb = t.64
	t.65 = root
	t.66 = 24
	t.67 = t.66
	if t.65 goto :null13
		Error("null pointer")
	null13:
	t.68 = [t.65]
	t.68 = [t.68+68]
	t.69 = call t.68(t.65 t.67)
	t.70 = t.69
	PrintIntS(t.70)
	t.71 = root
	t.72 = 12
	t.73 = t.72
	if t.71 goto :null14
		Error("null pointer")
	null14:
	t.74 = [t.71]
	t.74 = [t.74+68]
	t.75 = call t.74(t.71 t.73)
	t.76 = t.75
	PrintIntS(t.76)
	t.77 = root
	t.78 = 16
	t.79 = t.78
	if t.77 goto :null15
		Error("null pointer")
	null15:
	t.80 = [t.77]
	t.80 = [t.80+68]
	t.81 = call t.80(t.77 t.79)
	t.82 = t.81
	PrintIntS(t.82)
	t.83 = root
	t.84 = 50
	t.85 = t.84
	if t.83 goto :null16
		Error("null pointer")
	null16:
	t.86 = [t.83]
	t.86 = [t.86+68]
	t.87 = call t.86(t.83 t.85)
	t.88 = t.87
	PrintIntS(t.88)
	t.89 = root
	t.90 = 12
	t.91 = t.90
	if t.89 goto :null17
		Error("null pointer")
	null17:
	t.92 = [t.89]
	t.92 = [t.92+68]
	t.93 = call t.92(t.89 t.91)
	t.94 = t.93
	PrintIntS(t.94)
	t.95 = root
	t.96 = 12
	t.97 = t.96
	if t.95 goto :null18
		Error("null pointer")
	null18:
	t.98 = [t.95]
	t.98 = [t.98+52]
	t.99 = call t.98(t.95 t.97)
	t.100 = t.99
	ntb = t.100
	t.101 = root
	if t.101 goto :null19
		Error("null pointer")
	null19:
	t.102 = [t.101]
	t.102 = [t.102+72]
	t.103 = call t.102(t.101)
	t.104 = t.103
	ntb = t.104
	t.105 = root
	t.106 = 12
	t.107 = t.106
	if t.105 goto :null20
		Error("null pointer")
	null20:
	t.108 = [t.105]
	t.108 = [t.108+68]
	t.109 = call t.108(t.105 t.107)
	t.110 = t.109
	PrintIntS(t.110)
	t.0 = 0
	t.1 = t.0
	ret t.1

func Tree.Init(this v_key)
	t.2 = v_key
	t.3 = t.2
	[this+12] = t.3
	t.4 = 0
	t.5 = t.4
	[this+16] = t.5
	t.6 = 0
	t.7 = t.6
	[this+20] = t.7
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.SetRight(this rn)
	t.2 = rn
	t.3 = t.2
	[this+8] = t.3
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.SetLeft(this ln)
	t.2 = ln
	t.3 = t.2
	[this+4] = t.3
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.GetRight(this)
	t.0 = [this+8]
	t.1 = t.0
	ret t.1

func Tree.GetLeft(this)
	t.0 = [this+4]
	t.1 = t.0
	ret t.1

func Tree.GetKey(this)
	t.0 = [this+12]
	t.1 = t.0
	ret t.1

func Tree.SetKey(this v_key)
	t.2 = v_key
	t.3 = t.2
	[this+12] = t.3
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.GetHas_Right(this)
	t.0 = [this+20]
	t.1 = t.0
	ret t.1

func Tree.GetHas_Left(this)
	t.0 = [this+16]
	t.1 = t.0
	ret t.1

func Tree.SetHas_Left(this val)
	t.2 = val
	t.3 = t.2
	[this+16] = t.3
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.SetHas_Right(this val)
	t.2 = val
	t.3 = t.2
	[this+20] = t.3
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.Compare(this num1 num2)
	t.2 = 0
	t.3 = t.2
	ntb = t.3
	t.4 = num2
	t.5 = 1
	t.6 = Add(t.4 t.5)
	t.7 = t.6
	nti = t.7
	t.10 = num1
	t.11 = num2
	t.8 = t.10
	t.9 = t.11
	t.12 = LtS(t.8 t.9)
	t.13 = t.12
	if0 t.13 goto :if1_else
		t.14 = 0
		t.15 = t.14
		ntb = t.15
		goto :if1_end
	if1_else:
		t.19 = num1
		t.20 = nti
		t.17 = t.19
		t.18 = t.20
		t.21 = LtS(t.17 t.18)
		t.22 = t.21
		t.23 = t.22
		t.16 = t.23
		t.24 = Sub(1 t.16)
		t.25 = t.24
		t.26 = t.25
		t.27 = t.26
		if0 t.27 goto :if2_else
			t.28 = 0
			t.29 = t.28
			ntb = t.29
			goto :if2_end
		if2_else:
			t.30 = 1
			t.31 = t.30
			ntb = t.31
		if2_end:
	if1_end:
	t.0 = ntb
	t.1 = t.0
	ret t.1

func Tree.Insert(this v_key)
	t.2 = HeapAllocZ(28)
	[t.2] = :vmt_Tree
	t.3 = t.2
	t.4 = t.3
	new_node = t.4
	t.5 = new_node
	t.6 = v_key
	t.7 = t.6
	if t.5 goto :null21
		Error("null pointer")
	null21:
	t.8 = [t.5]
	t.8 = [t.8+0]
	t.9 = call t.8(t.5 t.7)
	t.10 = t.9
	ntb = t.10
	t.11 = this
	t.12 = t.11
	current_node = t.12
	t.13 = 1
	t.14 = t.13
	cont = t.14
	while1_top:
	t.15 = cont
	t.16 = t.15
	if0 t.16 goto :while1_end
		t.17 = current_node
		if t.17 goto :null22
			Error("null pointer")
		null22:
		t.18 = [t.17]
		t.18 = [t.18+20]
		t.19 = call t.18(t.17)
		t.20 = t.19
		key_aux = t.20
		t.23 = v_key
		t.24 = key_aux
		t.21 = t.23
		t.22 = t.24
		t.25 = LtS(t.21 t.22)
		t.26 = t.25
		if0 t.26 goto :if3_else
			t.27 = current_node
			if t.27 goto :null23
				Error("null pointer")
			null23:
			t.28 = [t.27]
			t.28 = [t.28+32]
			t.29 = call t.28(t.27)
			t.30 = t.29
			t.31 = t.30
			if0 t.31 goto :if4_else
				t.32 = current_node
				if t.32 goto :null24
					Error("null pointer")
				null24:
				t.33 = [t.32]
				t.33 = [t.33+16]
				t.34 = call t.33(t.32)
				t.35 = t.34
				current_node = t.35
				goto :if4_end
			if4_else:
				t.36 = 0
				t.37 = t.36
				cont = t.37
				t.38 = current_node
				t.39 = 1
				t.40 = t.39
				if t.38 goto :null25
					Error("null pointer")
				null25:
				t.41 = [t.38]
				t.41 = [t.41+36]
				t.42 = call t.41(t.38 t.40)
				t.43 = t.42
				ntb = t.43
				t.44 = current_node
				t.45 = new_node
				t.46 = t.45
				if t.44 goto :null26
					Error("null pointer")
				null26:
				t.47 = [t.44]
				t.47 = [t.47+8]
				t.48 = call t.47(t.44 t.46)
				t.49 = t.48
				ntb = t.49
			if4_end:
			goto :if3_end
		if3_else:
			t.50 = current_node
			if t.50 goto :null27
				Error("null pointer")
			null27:
			t.51 = [t.50]
			t.51 = [t.51+28]
			t.52 = call t.51(t.50)
			t.53 = t.52
			t.54 = t.53
			if0 t.54 goto :if5_else
				t.55 = current_node
				if t.55 goto :null28
					Error("null pointer")
				null28:
				t.56 = [t.55]
				t.56 = [t.56+12]
				t.57 = call t.56(t.55)
				t.58 = t.57
				current_node = t.58
				goto :if5_end
			if5_else:
				t.59 = 0
				t.60 = t.59
				cont = t.60
				t.61 = current_node
				t.62 = 1
				t.63 = t.62
				if t.61 goto :null29
					Error("null pointer")
				null29:
				t.64 = [t.61]
				t.64 = [t.64+40]
				t.65 = call t.64(t.61 t.63)
				t.66 = t.65
				ntb = t.66
				t.67 = current_node
				t.68 = new_node
				t.69 = t.68
				if t.67 goto :null30
					Error("null pointer")
				null30:
				t.70 = [t.67]
				t.70 = [t.70+4]
				t.71 = call t.70(t.67 t.69)
				t.72 = t.71
				ntb = t.72
			if5_end:
		if3_end:
		goto :while1_top
	while1_end:
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.Delete(this v_key)
	t.2 = this
	t.3 = t.2
	current_node = t.3
	t.4 = this
	t.5 = t.4
	parent_node = t.5
	t.6 = 1
	t.7 = t.6
	cont = t.7
	t.8 = 0
	t.9 = t.8
	found = t.9
	t.10 = 1
	t.11 = t.10
	is_root = t.11
	while2_top:
	t.12 = cont
	t.13 = t.12
	if0 t.13 goto :while2_end
		t.14 = current_node
		if t.14 goto :null31
			Error("null pointer")
		null31:
		t.15 = [t.14]
		t.15 = [t.15+20]
		t.16 = call t.15(t.14)
		t.17 = t.16
		key_aux = t.17
		t.20 = v_key
		t.21 = key_aux
		t.18 = t.20
		t.19 = t.21
		t.22 = LtS(t.18 t.19)
		t.23 = t.22
		if0 t.23 goto :if6_else
			t.24 = current_node
			if t.24 goto :null32
				Error("null pointer")
			null32:
			t.25 = [t.24]
			t.25 = [t.25+32]
			t.26 = call t.25(t.24)
			t.27 = t.26
			t.28 = t.27
			if0 t.28 goto :if7_else
				t.29 = current_node
				t.30 = t.29
				parent_node = t.30
				t.31 = current_node
				if t.31 goto :null33
					Error("null pointer")
				null33:
				t.32 = [t.31]
				t.32 = [t.32+16]
				t.33 = call t.32(t.31)
				t.34 = t.33
				current_node = t.34
				goto :if7_end
			if7_else:
				t.35 = 0
				t.36 = t.35
				cont = t.36
			if7_end:
			goto :if6_end
		if6_else:
			t.39 = key_aux
			t.40 = v_key
			t.37 = t.39
			t.38 = t.40
			t.41 = LtS(t.37 t.38)
			t.42 = t.41
			if0 t.42 goto :if8_else
				t.43 = current_node
				if t.43 goto :null34
					Error("null pointer")
				null34:
				t.44 = [t.43]
				t.44 = [t.44+28]
				t.45 = call t.44(t.43)
				t.46 = t.45
				t.47 = t.46
				if0 t.47 goto :if9_else
					t.48 = current_node
					t.49 = t.48
					parent_node = t.49
					t.50 = current_node
					if t.50 goto :null35
						Error("null pointer")
					null35:
					t.51 = [t.50]
					t.51 = [t.51+12]
					t.52 = call t.51(t.50)
					t.53 = t.52
					current_node = t.53
					goto :if9_end
				if9_else:
					t.54 = 0
					t.55 = t.54
					cont = t.55
				if9_end:
				goto :if8_end
			if8_else:
				t.56 = is_root
				t.57 = t.56
				t.58 = t.57
				if0 t.58 goto :if10_else
					t.61 = current_node
					if t.61 goto :null36
						Error("null pointer")
					null36:
					t.62 = [t.61]
					t.62 = [t.62+28]
					t.63 = call t.62(t.61)
					t.64 = t.63
					t.60 = t.64
					t.65 = Sub(1 t.60)
					t.66 = t.65
					t.67 = t.66
					t.68 = t.67
					if0 t.68 goto :ss1_else
						t.70 = current_node
						if t.70 goto :null37
							Error("null pointer")
						null37:
						t.71 = [t.70]
						t.71 = [t.71+32]
						t.72 = call t.71(t.70)
						t.73 = t.72
						t.69 = t.73
						t.74 = Sub(1 t.69)
						t.75 = t.74
						t.76 = t.75
						t.77 = t.76
					goto :ss1_end
					ss1_else:
					t.59 = 0
					ss1_end:
					t.78 = t.59
					t.79 = t.78
					if0 t.79 goto :if11_else
						t.80 = 1
						t.81 = t.80
						ntb = t.81
						goto :if11_end
					if11_else:
						t.82 = this
						t.83 = parent_node
						t.84 = t.83
						t.85 = current_node
						t.86 = t.85
						if t.82 goto :null38
							Error("null pointer")
						null38:
						t.87 = [t.82]
						t.87 = [t.87+56]
						t.88 = call t.87(t.82 t.84 t.86)
						t.89 = t.88
						ntb = t.89
					if11_end:
					goto :if10_end
				if10_else:
					t.90 = this
					t.91 = parent_node
					t.92 = t.91
					t.93 = current_node
					t.94 = t.93
					if t.90 goto :null39
						Error("null pointer")
					null39:
					t.95 = [t.90]
					t.95 = [t.95+56]
					t.96 = call t.95(t.90 t.92 t.94)
					t.97 = t.96
					ntb = t.97
				if10_end:
				t.98 = 1
				t.99 = t.98
				found = t.99
				t.100 = 0
				t.101 = t.100
				cont = t.101
			if8_end:
		if6_end:
		t.102 = 0
		t.103 = t.102
		is_root = t.103
		goto :while2_top
	while2_end:
	t.0 = found
	t.1 = t.0
	ret t.1

func Tree.Remove(this c_node p_node)
	t.2 = c_node
	if t.2 goto :null40
		Error("null pointer")
	null40:
	t.3 = [t.2]
	t.3 = [t.3+32]
	t.4 = call t.3(t.2)
	t.5 = t.4
	t.6 = t.5
	if0 t.6 goto :if12_else
		t.7 = this
		t.8 = p_node
		t.9 = t.8
		t.10 = c_node
		t.11 = t.10
		if t.7 goto :null41
			Error("null pointer")
		null41:
		t.12 = [t.7]
		t.12 = [t.12+64]
		t.13 = call t.12(t.7 t.9 t.11)
		t.14 = t.13
		ntb = t.14
		goto :if12_end
	if12_else:
		t.15 = c_node
		if t.15 goto :null42
			Error("null pointer")
		null42:
		t.16 = [t.15]
		t.16 = [t.16+28]
		t.17 = call t.16(t.15)
		t.18 = t.17
		t.19 = t.18
		if0 t.19 goto :if13_else
			t.20 = this
			t.21 = p_node
			t.22 = t.21
			t.23 = c_node
			t.24 = t.23
			if t.20 goto :null43
				Error("null pointer")
			null43:
			t.25 = [t.20]
			t.25 = [t.25+60]
			t.26 = call t.25(t.20 t.22 t.24)
			t.27 = t.26
			ntb = t.27
			goto :if13_end
		if13_else:
			t.28 = c_node
			if t.28 goto :null44
				Error("null pointer")
			null44:
			t.29 = [t.28]
			t.29 = [t.29+20]
			t.30 = call t.29(t.28)
			t.31 = t.30
			auxkey1 = t.31
			t.32 = p_node
			if t.32 goto :null45
				Error("null pointer")
			null45:
			t.33 = [t.32]
			t.33 = [t.33+16]
			t.34 = call t.33(t.32)
			t.35 = t.34
			t.36 = t.35
			if t.36 goto :null46
				Error("null pointer")
			null46:
			t.37 = [t.36]
			t.37 = [t.37+20]
			t.38 = call t.37(t.36)
			t.39 = t.38
			auxkey2 = t.39
			t.40 = this
			t.41 = auxkey1
			t.42 = t.41
			t.43 = auxkey2
			t.44 = t.43
			if t.40 goto :null47
				Error("null pointer")
			null47:
			t.45 = [t.40]
			t.45 = [t.45+44]
			t.46 = call t.45(t.40 t.42 t.44)
			t.47 = t.46
			t.48 = t.47
			if0 t.48 goto :if14_else
				t.49 = p_node
				t.50 = [this+24]
				t.51 = t.50
				if t.49 goto :null48
					Error("null pointer")
				null48:
				t.52 = [t.49]
				t.52 = [t.52+8]
				t.53 = call t.52(t.49 t.51)
				t.54 = t.53
				ntb = t.54
				t.55 = p_node
				t.56 = 0
				t.57 = t.56
				if t.55 goto :null49
					Error("null pointer")
				null49:
				t.58 = [t.55]
				t.58 = [t.58+36]
				t.59 = call t.58(t.55 t.57)
				t.60 = t.59
				ntb = t.60
				goto :if14_end
			if14_else:
				t.61 = p_node
				t.62 = [this+24]
				t.63 = t.62
				if t.61 goto :null50
					Error("null pointer")
				null50:
				t.64 = [t.61]
				t.64 = [t.64+4]
				t.65 = call t.64(t.61 t.63)
				t.66 = t.65
				ntb = t.66
				t.67 = p_node
				t.68 = 0
				t.69 = t.68
				if t.67 goto :null51
					Error("null pointer")
				null51:
				t.70 = [t.67]
				t.70 = [t.70+40]
				t.71 = call t.70(t.67 t.69)
				t.72 = t.71
				ntb = t.72
			if14_end:
		if13_end:
	if12_end:
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.RemoveRight(this c_node p_node)
	while3_top:
	t.2 = c_node
	if t.2 goto :null52
		Error("null pointer")
	null52:
	t.3 = [t.2]
	t.3 = [t.3+28]
	t.4 = call t.3(t.2)
	t.5 = t.4
	if0 t.5 goto :while3_end
		t.6 = c_node
		t.7 = c_node
		if t.7 goto :null53
			Error("null pointer")
		null53:
		t.8 = [t.7]
		t.8 = [t.8+12]
		t.9 = call t.8(t.7)
		t.10 = t.9
		t.11 = t.10
		if t.11 goto :null54
			Error("null pointer")
		null54:
		t.12 = [t.11]
		t.12 = [t.12+20]
		t.13 = call t.12(t.11)
		t.14 = t.13
		if t.6 goto :null55
			Error("null pointer")
		null55:
		t.15 = [t.6]
		t.15 = [t.15+24]
		t.16 = call t.15(t.6 t.14)
		t.17 = t.16
		ntb = t.17
		t.18 = c_node
		t.19 = t.18
		p_node = t.19
		t.20 = c_node
		if t.20 goto :null56
			Error("null pointer")
		null56:
		t.21 = [t.20]
		t.21 = [t.21+12]
		t.22 = call t.21(t.20)
		t.23 = t.22
		c_node = t.23
		goto :while3_top
	while3_end:
	t.24 = p_node
	t.25 = [this+24]
	t.26 = t.25
	if t.24 goto :null57
		Error("null pointer")
	null57:
	t.27 = [t.24]
	t.27 = [t.27+4]
	t.28 = call t.27(t.24 t.26)
	t.29 = t.28
	ntb = t.29
	t.30 = p_node
	t.31 = 0
	t.32 = t.31
	if t.30 goto :null58
		Error("null pointer")
	null58:
	t.33 = [t.30]
	t.33 = [t.33+40]
	t.34 = call t.33(t.30 t.32)
	t.35 = t.34
	ntb = t.35
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.RemoveLeft(this c_node p_node)
	while4_top:
	t.2 = c_node
	if t.2 goto :null59
		Error("null pointer")
	null59:
	t.3 = [t.2]
	t.3 = [t.3+32]
	t.4 = call t.3(t.2)
	t.5 = t.4
	if0 t.5 goto :while4_end
		t.6 = c_node
		t.7 = c_node
		if t.7 goto :null60
			Error("null pointer")
		null60:
		t.8 = [t.7]
		t.8 = [t.8+16]
		t.9 = call t.8(t.7)
		t.10 = t.9
		t.11 = t.10
		if t.11 goto :null61
			Error("null pointer")
		null61:
		t.12 = [t.11]
		t.12 = [t.12+20]
		t.13 = call t.12(t.11)
		t.14 = t.13
		if t.6 goto :null62
			Error("null pointer")
		null62:
		t.15 = [t.6]
		t.15 = [t.15+24]
		t.16 = call t.15(t.6 t.14)
		t.17 = t.16
		ntb = t.17
		t.18 = c_node
		t.19 = t.18
		p_node = t.19
		t.20 = c_node
		if t.20 goto :null63
			Error("null pointer")
		null63:
		t.21 = [t.20]
		t.21 = [t.21+16]
		t.22 = call t.21(t.20)
		t.23 = t.22
		c_node = t.23
		goto :while4_top
	while4_end:
	t.24 = p_node
	t.25 = [this+24]
	t.26 = t.25
	if t.24 goto :null64
		Error("null pointer")
	null64:
	t.27 = [t.24]
	t.27 = [t.27+8]
	t.28 = call t.27(t.24 t.26)
	t.29 = t.28
	ntb = t.29
	t.30 = p_node
	t.31 = 0
	t.32 = t.31
	if t.30 goto :null65
		Error("null pointer")
	null65:
	t.33 = [t.30]
	t.33 = [t.33+36]
	t.34 = call t.33(t.30 t.32)
	t.35 = t.34
	ntb = t.35
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.Search(this v_key)
	t.2 = this
	t.3 = t.2
	current_node = t.3
	t.4 = 1
	t.5 = t.4
	cont = t.5
	t.6 = 0
	t.7 = t.6
	ifound = t.7
	while5_top:
	t.8 = cont
	t.9 = t.8
	if0 t.9 goto :while5_end
		t.10 = current_node
		if t.10 goto :null66
			Error("null pointer")
		null66:
		t.11 = [t.10]
		t.11 = [t.11+20]
		t.12 = call t.11(t.10)
		t.13 = t.12
		key_aux = t.13
		t.16 = v_key
		t.17 = key_aux
		t.14 = t.16
		t.15 = t.17
		t.18 = LtS(t.14 t.15)
		t.19 = t.18
		if0 t.19 goto :if15_else
			t.20 = current_node
			if t.20 goto :null67
				Error("null pointer")
			null67:
			t.21 = [t.20]
			t.21 = [t.21+32]
			t.22 = call t.21(t.20)
			t.23 = t.22
			t.24 = t.23
			if0 t.24 goto :if16_else
				t.25 = current_node
				if t.25 goto :null68
					Error("null pointer")
				null68:
				t.26 = [t.25]
				t.26 = [t.26+16]
				t.27 = call t.26(t.25)
				t.28 = t.27
				current_node = t.28
				goto :if16_end
			if16_else:
				t.29 = 0
				t.30 = t.29
				cont = t.30
			if16_end:
			goto :if15_end
		if15_else:
			t.33 = key_aux
			t.34 = v_key
			t.31 = t.33
			t.32 = t.34
			t.35 = LtS(t.31 t.32)
			t.36 = t.35
			if0 t.36 goto :if17_else
				t.37 = current_node
				if t.37 goto :null69
					Error("null pointer")
				null69:
				t.38 = [t.37]
				t.38 = [t.38+28]
				t.39 = call t.38(t.37)
				t.40 = t.39
				t.41 = t.40
				if0 t.41 goto :if18_else
					t.42 = current_node
					if t.42 goto :null70
						Error("null pointer")
					null70:
					t.43 = [t.42]
					t.43 = [t.43+12]
					t.44 = call t.43(t.42)
					t.45 = t.44
					current_node = t.45
					goto :if18_end
				if18_else:
					t.46 = 0
					t.47 = t.46
					cont = t.47
				if18_end:
				goto :if17_end
			if17_else:
				t.48 = 1
				t.49 = t.48
				ifound = t.49
				t.50 = 0
				t.51 = t.50
				cont = t.51
			if17_end:
		if15_end:
		goto :while5_top
	while5_end:
	t.0 = ifound
	t.1 = t.0
	ret t.1

func Tree.Print(this)
	t.2 = this
	t.3 = t.2
	current_node = t.3
	t.4 = this
	t.5 = current_node
	t.6 = t.5
	if t.4 goto :null71
		Error("null pointer")
	null71:
	t.7 = [t.4]
	t.7 = [t.7+76]
	t.8 = call t.7(t.4 t.6)
	t.9 = t.8
	ntb = t.9
	t.0 = 1
	t.1 = t.0
	ret t.1

func Tree.RecPrint(this node)
	t.2 = node
	if t.2 goto :null72
		Error("null pointer")
	null72:
	t.3 = [t.2]
	t.3 = [t.3+32]
	t.4 = call t.3(t.2)
	t.5 = t.4
	t.6 = t.5
	if0 t.6 goto :if19_else
		t.7 = this
		t.8 = node
		if t.8 goto :null73
			Error("null pointer")
		null73:
		t.9 = [t.8]
		t.9 = [t.9+16]
		t.10 = call t.9(t.8)
		t.11 = t.10
		if t.7 goto :null74
			Error("null pointer")
		null74:
		t.12 = [t.7]
		t.12 = [t.12+76]
		t.13 = call t.12(t.7 t.11)
		t.14 = t.13
		ntb = t.14
		goto :if19_end
	if19_else:
		t.15 = 1
		t.16 = t.15
		ntb = t.16
	if19_end:
	t.17 = node
	if t.17 goto :null75
		Error("null pointer")
	null75:
	t.18 = [t.17]
	t.18 = [t.18+20]
	t.19 = call t.18(t.17)
	t.20 = t.19
	PrintIntS(t.20)
	t.21 = node
	if t.21 goto :null76
		Error("null pointer")
	null76:
	t.22 = [t.21]
	t.22 = [t.22+28]
	t.23 = call t.22(t.21)
	t.24 = t.23
	t.25 = t.24
	if0 t.25 goto :if20_else
		t.26 = this
		t.27 = node
		if t.27 goto :null77
			Error("null pointer")
		null77:
		t.28 = [t.27]
		t.28 = [t.28+12]
		t.29 = call t.28(t.27)
		t.30 = t.29
		if t.26 goto :null78
			Error("null pointer")
		null78:
		t.31 = [t.26]
		t.31 = [t.31+76]
		t.32 = call t.31(t.26 t.30)
		t.33 = t.32
		ntb = t.33
		goto :if20_end
	if20_else:
		t.34 = 1
		t.35 = t.34
		ntb = t.35
	if20_end:
	t.0 = 1
	t.1 = t.0
	ret t.1

